Source/WebCore/rendering/svg/SVGTextQuery.cpp

```diff
@@ -137,63 +137,53 @@ bool SVGTextQuery::executeQuery(Data* queryData, ProcessTextFragmentCallback fra
 
 bool SVGTextQuery::mapStartEndPositionsIntoFragmentCoordinates(Data* queryData, const SVGTextFragment& fragment, unsigned& startPosition, unsigned& endPosition) const
 {
-    // Reuse the same logic used for text selection & painting, to map our query start/length into start/endPositions of the current text fragment.
+    // Make <startPosition, endPosition> offsets relative to the current text box.
     ASSERT(startPosition >= queryData->processedCharacters);
     ASSERT(endPosition >= queryData->processedCharacters);
-    startPosition -= queryData->processedCharacters;
-    endPosition -= queryData->processedCharacters;
 
-    // <startPosition, endPosition> is now a tuple of offsets relative to the current text box.
-    // Compute the offsets of the fragment in the same offset space.
-    unsigned fragmentStartInBox = fragment.characterOffset - queryData->textBox->start();
-    unsigned fragmentEndInBox = fragmentStartInBox + fragment.length;
+    unsigned boxStart = queryData->processedCharacters;
 
-    // Check if the ranges intersect.
-    startPosition = std::max<unsigned>(startPosition, fragmentStartInBox);
-    endPosition = std::min<unsigned>(endPosition, fragmentEndInBox);
+    startPosition -= boxStart;
+    endPosition -= boxStart;
```

Comment:
This will map the `startPosition` and `endPosition` from the _Line coordinates_ to the _InlineTextBox coordinates_.